### PR TITLE
[WIP] Creating Aurora MySQL instance for ACL backend

### PR DESCRIPTION
**Pull request by rajeevr2715**
**Requirements**
Create a AWS managed service - aurora provisioned  DB using MySQL Engine

**Description**
1. Created the provisioned DB
2. Set the credentials in Jenkins
3. Update the ACL cluster and instance details in Installer vars json file.

### CODE REVIEW

File: installscripts/jazz-terraform-unix-noinstances/acl.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
+resource "aws_rds_cluster" "casbin" {
+  cluster_identifier      = "casbin-cluster"
+  availability_zones      = ["us-east-1a", "us-east-1b"]
+  database_name           = "casbin"
+  master_username         = "root"
+  master_password         = "casbin2715"
+  backup_retention_period = 5
+  preferred_backup_window = "07:00-09:00"
+  skip_final_snapshot = true
+  engine                  = "aurora-mysql"
```
Comment (svidhani) [inline installscripts/jazz-terraform-unix-noinstances/acl.tf]: i dont see the mysql version (5.6, 5.7). Are you going with default?

Comment (rajeevr2715) [reply installscripts/jazz-terraform-unix-noinstances/acl.tf]: "engine". https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Updates.20180206.html#AuroraMySQL.Updates.20180206.CLI

File: installscripts/jazz-terraform-unix-noinstances/acl.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
+resource "aws_rds_cluster" "casbin" {
+  cluster_identifier      = "casbin-cluster"
+  availability_zones      = ["us-east-1a", "us-east-1b"]
+  database_name           = "casbin"
+  master_username         = "root"
+  master_password         = "casbin2715"
```
[the remark this answers is not in the document]
Comment (rajeevr2715) [reply installscripts/jazz-terraform-unix-noinstances/acl.tf]: Done.

File: installscripts/jazz-terraform-unix-noinstances/acl.tf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        from_port = 3306
+        to_port = 3306
+        protocol = "tcp"
+        self = true
+    }    
+    egress {
+        from_port = 0
+        to_port = 0
+        protocol = "-1"
+        cidr_blocks = ["0.0.0.0/0"]
```
[the remark this answers is not in the document]
Comment (rajeevr2715) [reply installscripts/jazz-terraform-unix-noinstances/acl.tf]: Done.

File: installscripts/jazz-terraform-unix-noinstances/acl.tf (hunk 1 above)
Comment (rajeevr2715) [reply installscripts/jazz-terraform-unix-noinstances/acl.tf]: Updated the version as well.

File: installscripts/cookbooks/jenkins/attributes/default.rb [hunk 4]
```diff
@@ -25,6 +25,8 @@
 default['bbpassword'] = 'REPLACEME'
 default['sonaruser'] = 'REPLACEME'
 default['sonarpassword'] = 'REPLACEME'
+default['acluser'] = 'REPLACEME'
```
Comment (devsatishm) [inline installscripts/cookbooks/jenkins/attributes/default.rb]: `acluster` is little confusing. Can we use `acl_db_cluster` & `acl_db_password`?

File: installscripts/cookbooks/jenkins/files/credentials/acl.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+JENKINS_CLI_CMD=$1
+ACLUSER=$2
+ACLPASSWORD=$3
+
+cat <<EOF | $JENKINS_CLI_CMD create-credentials-by-xml system::system::jenkins "(global)"
+<com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+  <scope>GLOBAL</scope>
+  <id>ACL_CRDENTIAL</id>
```
Comment (devsatishm) [inline installscripts/cookbooks/jenkins/files/credentials/acl.sh]: `ACL_DB_CREDENTIAL`?

File: installscripts/cookbooks/jenkins/files/default/jazz-installer-vars.json [hunk 6]
```diff
@@ -112,5 +112,13 @@
         "SLACK_TOKEN": "{slack_token}",
         "SLACK_WORKSPACE": "{slack_workspace}",
         "SLACK_SVC_ID": "{slack_svc_acc_id}"
+    },
+    "ACL": {
```
Comment (devsatishm) [inline installscripts/cookbooks/jenkins/files/default/jazz-installer-vars.json]: Can we use the schema below? Basically, extending what you've got already to support other related configurations later. Set `ENABLED = true` by default.

```json
"ACL": {
  "ENABLED": true,
  "DATABASE": {
    "CLUSTER_READER_ENDPOINT": "",
    "CLUSTER_WRITER_ENDPOINT": "",
    "ENDPOINT": "",
    "PORT": "",
    "NAME": "",
    "CREDENTIAL_ID": ""
  }
}
```

File: installscripts/jazz-terraform-unix-noinstances/acl.tf [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  identifier              = "casbin"
+  instance_class          = "db.t2.small"
+  engine                  = "aurora-mysql"
+  engine_version          = "5.7.12"
+  publicly_accessible     = true
+
+  tags = "${merge(var.additional_tags, local.common_tags)}"
+}
+
+# TO DO  - to remove in next version
```
Comment (devsatishm) [inline installscripts/jazz-terraform-unix-noinstances/acl.tf]: Sorry, I was referring to `cidr_blocks` specifically.  You can say - `#TODO:  CIDR  block will be refined/more restricted in the next version`

File: installscripts/cookbooks/jenkins/files/credentials/acl.sh (hunk 5 above)
Comment (rajeevr2715) [reply installscripts/cookbooks/jenkins/files/credentials/acl.sh]: yes

File: installscripts/cookbooks/jenkins/files/default/jazz-installer-vars.json (hunk 6 above)
Comment (rajeevr2715) [reply installscripts/cookbooks/jenkins/files/default/jazz-installer-vars.json]: Done

File: installscripts/jazz-terraform-unix-noinstances/acl.tf (hunk 7 above)
Comment (rajeevr2715) [reply installscripts/jazz-terraform-unix-noinstances/acl.tf]: Done

File: installscripts/cookbooks/jenkins/attributes/default.rb (hunk 4 above)
Comment (rajeevr2715) [reply installscripts/cookbooks/jenkins/attributes/default.rb]: Done

**Pull request merged**